doc/develop/modules.rst

```diff
@@ -86,19 +86,50 @@ Module Repositories
 
 * Module repositories names should be explicitly set in the :file:`zephyr/module.yml` file.
 
-* Modules should use "zephyr" as the default name for the repository main
-  branch. Branches for specific purposes, for example, a module branch for
-  an LTS Zephyr version, shall have names starting with the 'zephyr\_' prefix.
+* If the module has an external (upstream) project repository:
 
-* If the module has an external (upstream) project repository, the module
-  repository should preserve the upstream repository folder structure.
+  * The module repository should preserve the upstream repository folder structure.
+
+  * Follow one of these strategies to track the upstream repository:
+
+    * The preferred strategy is that the module tracks directly the upstream main development
+      branch or maintenance branches which are pointed by the Zephyr manifest without any changes,
+      where all Zephyr specific code is hosted by the upstream project.
+      Pull requests are sent directly to the upstream repository.
```

Comment:
> When I maintain a subsystem/driver, I definitely do not want to be forced, or force a collaborator, to open a PR upstream for a HAL/optional module.

Well, if the upstream is responsive and well behaved it would not need to be a problem.
But if it would not be, the solution would be to create a `zephyr_` branch until the fix comes back from upstream. It would be the responsibility of the module maintainer to ensure developers in Zephyr can work.

Note that the idea here is not that for something which is as much behind the upstream as today is, say, the nordic hal module, one would need to submit upstream and wait 2 months for it to come back to the downstream.
The idea is for modules whose main development branch would be synch'ed from the upstream either automatically in a matter of minutes, or by the maintainer in a matter of hours.